@raybejjani Getting better...

```
test $ CNI_INTEGRATION=gke K8S_VERSION=1.13 CILIUM_IMAGE="quay.io/cilium/cilium:latest" CILIUM_OPERATOR_IMAGE="quay.io/cilium/operator:latest" ginkgo --focus="K8sHealth*" -noColor -- -cilium.provision=false -cilium.kub
econfig=~/.kube/config -cilium.image="quay.io/cilium/cilium:latest" -cilium.operator-image="quay.io/cilium/operator:latest" -cilium.passCLIEnvironment=true
Running Suite: Suite-k8s-1.13
=============================
Random Seed: 1580844554
Will run 1 of 377 specs

Spinning up vagrant VMs... ●SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

------------------------------
• Failure in Spec Setup (BeforeEach) [0.504 seconds]
K8sHealthTest [BeforeEach]
/home/joe/work/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:395
  checks cilium-health status between nodes
  /home/joe/work/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:430

  Cannot connect to k8s cluster, output:
  W0204 11:29:28.665237   10845 loader.go:223] Config not found: ~/.kube/config
  The connection to the server localhost:8080 was refused - did you specify the right host or port?
  

  /home/joe/work/src/github.com/cilium/cilium/test/test_suite_test.go:276
------------------------------
<Checks>

</Checks>

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:

[Fail] [BeforeEach] K8sHealthTest checks cilium-health status between nodes 
/home/joe/work/src/github.com/cilium/cilium/test/test_suite_test.go:276

Ran 1 of 377 Specs in 0.514 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 376 Skipped
--- FAIL: TestTest (0.52s)
FAIL

Ginkgo ran 1 suite in 14.730666295s
Test Suite Failed
```

--- a/Documentation/contributing/testing/e2e.rst
+++ b/Documentation/contributing/testing/e2e.rst
@@ -440,9 +440,10 @@ This mode expects:
 - A populated K8S_VERSION environment variable set to the version of the cluster
 
 - If appropriate, set the ``CNI_INTEGRATION`` environment variable set to one
-  of ``flannel``, ``eks``, ``microk8s`` or ``minikube``. This selects matching
-  configuration overrides for cilium.
+  of ``flannel``, ``gke``, ``eks``, ``microk8s`` or ``minikube``. This selects
+  matching configuration overrides for cilium.
   Leaving this unset for non-matching integrations is also correct.
+
   For k8s environments that invoke an authentication agent, such as EKS and
   ``aws-iam-authenticator``, set ``--cilium.passCLIEnvironment=true``
 
@@ -450,8 +451,37 @@ An example invocation is
 
 ::
 
-  CNI_INTEGRATION=eks K8S_VERSION=1.13 CILIUM_IMAGE="quay.io/cilium/cilium:latest" CILIUM_OPERATOR_IMAGE="quay.io/cilium/operator:latest" ginkgo --focus="K8s*" -noColor -- -cilium.provision=false -cilium.kubeconfig=~/.kube/config -cilium.image="quay.io/cilium/cilium:latest" -cilium.operator-image="quay.io/cilium/operator:latest" -cilium.passCLIEnvironment=true
+  CNI_INTEGRATION=eks K8S_VERSION=1.13 CILIUM_IMAGE="quay.io/cilium/cilium:latest" CILIUM_OPERATOR_IMAGE="quay.io/cilium/operator:latest" ginkgo --focus="K8s*" -noColor -- -cilium.provision=false -cilium.kubeconfig=`echo ~/.kube/config` -cilium.image="quay.io/cilium/cilium:latest" -cilium.operator-image="quay.io/cilium/operator:latest" -cilium.passCLIEnvironment=true
 
+GKE (experimental)
+^^^^^^^^^^^^^^^^^^^^^^
+
+Not all tests can succeed on GKE. Many do, however and may be useful.
+
+1- Setup a cluster as in :ref:`k8s_install_gke` or utilize an existing
+cluster.
+
+.. note:: The tests require machines larger than ``n1-standard-4``. This can be
+          set with ``--machine-type n1-standard-4`` on cluster creation.
+
+
+2- Label 2 nodes for testing with ``cilium.io/ci-node=k8s1`` and
+``cilium.io/ci-node=k8s2``
+
+::
+
+  kubectl label node gke-my-cluster-default-pool-b011879a-6j26 cilium.io/ci-node=k8s1
+  kubectl label node gke-my-cluster-default-pool-b011879a-b1r2 cilium.io/ci-node=k8s2
+
+3- Invoke the tests from ``cilium/test`` with options set as explained in
+`Running End-To-End Tests In Other Environments via kubeconfig`_
+
+::
+
+  CNI_INTEGRATION=gke K8S_VERSION=1.13 CILIUM_IMAGE="quay.io/cilium/cilium:latest" CILIUM_OPERATOR_IMAGE="quay.io/cilium/operator:latest" ginkgo --focus="K8s*" -noColor -- -cilium.provision=false -cilium.kubeconfig=`echo ~/.kube/config` -cilium.image="quay.io/cilium/cilium:latest" -cilium.operator-image="quay.io/cilium/operator:latest" -cilium.passCLIEnvironment=true
+
+.. note:: The kubernetes version defaults to 1.13 but can be configured with
+          versions between 1.13 and 1.15. Check with ``kubectl version`` 
 
 AWS EKS (experimental)
 ^^^^^^^^^^^^^^^^^^^^^^
@@ -474,7 +504,7 @@ cluster.
 
 ::
 
-  CNI_INTEGRATION=eks K8S_VERSION=1.14  CILIUM_IMAGE="quay.io/cilium/cilium:latest" CILIUM_OPERATOR_IMAGE="quay.io/cilium/operator:latest" ginkgo --focus="K8s*" -noColor -- -cilium.provision=false -cilium.kubeconfig=~/.kube/config -cilium.image="quay.io/cilium/cilium:latest" -cilium.operator-image="quay.io/cilium/operator:latest" -cilium.passCLIEnvironment=true
+  CNI_INTEGRATION=eks K8S_VERSION=1.14 CILIUM_IMAGE="quay.io/cilium/cilium:latest" CILIUM_OPERATOR_IMAGE="quay.io/cilium/operator:latest" ginkgo --focus="K8s*" -noColor -- -cilium.provision=false -cilium.kubeconfig=`echo ~/.kube/config` -cilium.image="quay.io/cilium/cilium:latest" -cilium.operator-image="quay.io/cilium/operator:latest" -cilium.passCLIEnvironment=true
 
 Be sure to pass ``--cilium.passCLIEnvironment=true`` to allow kubectl to invoke ``aws-iam-authenticator``
 

--- a/Documentation/gettingstarted/k8s-install-gke.rst
+++ b/Documentation/gettingstarted/k8s-install-gke.rst
@@ -4,6 +4,8 @@
     Please use the official rendered version released here:
     http://docs.cilium.io
 
+.. _k8s_install_gke:
+
 **************************
 Installation on Google GKE
 **************************

